### ISSUE TITLE
New guides wanted

### ISSUE THREAD

**Issue by bvaughn** [reactjs/react.dev#83]
**Note** this issue has some overlap with FAQ section issue #29.

We revamped the docs, but there are a few guides that are missing and that I think are essential.

### Available
- [ ] @mateoholman Community section with good resources (e.g. https://github.com/facebook/react/pull/7117)
- [ ] @mrscobbler Glossary (similar to "Virtual DOM Terminology" that is now gone, but more friendly. Check http://redux.js.org/docs/Glossary.html for inspiration.)
- [ ] @chiangs Troubleshooting (there are ~5 common issues we should explain right away, like wrong element casing, not loaded DOM container, missing React in scope, etc. See http://redux.js.org/docs/Troubleshooting.html for inspiration)
- [ ] @madhug Manipulating Children (Using `cloneElement` and `Children` https://github.com/facebook/react/issues/6942)
- [ ] "Portals" (useful for modals)
- [ ] @mateoholman Animation with React Motion
- [ ] @jamesknelson Routing (no need to use React Router here, just show how to use `setState` with HTML5 history API and point to RR as a component-based API for it once v4 is out)
- [ ] @ishankbahl Snapshot testing (https://github.com/facebook/react/issues/7431)
- [ ] @mateoholman Publishing a Component (https://github.com/facebook/react/issues/7882)

### In progress
- [ ] @alexkrolick, @ishankbahl: AJAX (just show how to use `fetch()` in lifecycles, mention cancellation issues)

If somebody wants to work on this please comment here and let's figure out a plan. Help wanted!

Carry over from facebook/react/issues/8060

**Comment by markerikson** [comment]
The ReactPerf item may sorta be OBE, since it's not included in v16.

**Comment by bvaughn** [comment]
Not sure what that acronym means but I assume you're suggesting we should remove it? 😄 I don't object.

**Comment by JasonEtco** [comment]
Not sure if you're already planning it, but I would add **Didn't bind class methods** to common troubleshooting issues 👍

**Comment by markerikson** [comment]
@bvaughn : yeah, that's what I was suggesting :)

@JasonEtco : the "troubleshooting" suggestion overlaps heavily with the [proposal for an FAQ page](https://github.com/reactjs/reactjs.org/issues/29).

**Comment by AlmeroSteyn** [comment]
@bvaughn Accessibility has been added and can be checked off :-)

**Comment by mrscobbler** [comment]
I'll take the glossary that I created in the old repo and migrate it to the new docs repo

**Comment by alexkrolick** [comment]
As @markerikson mentioned there is some overlap with the FAQ discussion (#29). PR #43 covers AJAX and some of the VDOM concepts although a full-fledged glossary would probably replace that section.

**Comment by bvaughn** [comment]
Thanks @AlmeroSteyn. I've removed that entry.

Yeah, there's some overlap. I spent a few hours merging/moving issues and PRs yesterday but some of the larger/vaguer ones like this are a bit hard to do without a bit of overlap. I'll add a note in the description about the relationship.

**Comment by ishankbahl** [comment]
Hey @bvaughn , I would like to work on "AJAX (just show how to use `fetch()` in lifecycles, mention cancellation issues)". Though now we don't have to talk about cancellation issues as fetch now supports abort https://developers.google.com/web/updates/2017/09/abortable-fetch 
Can I take this one?

**Comment by bvaughn** [comment]
It's all yours, @ishankbahl

**Comment by alexkrolick** [comment]
@ishankbahl @bvaughn AJAX is already in the FAQ PR #43, as I pointed out.

If that is too big of a PR to review I can break it up, but let's not duplicate effort.

**Comment by bvaughn** [comment]
Sorry for missing that note, @alexkrolick. Thanks for pointing it out. Yeah, maybe hold off on doing anything with the AJAX section until after #43 has been reviewed @ishankbahl.

**Comment by mrscobbler** [comment]
I just created a pull request for the glossary content: https://github.com/reactjs/reactjs.org/pull/153

**Comment by damaera** [comment]
while in vue docs there's something like this.
![image](https://user-images.githubusercontent.com/5365957/31600350-4948a6d8-b280-11e7-92f2-e8c7f4af0edc.png)

I think routing should have their own menu, because almost all SPA have router.
state management too.

**Comment by mateoholman** [comment]
I'll take a stab at creating a guide for publishing a component if no one is currently working on it?

**Comment by bvaughn** [comment]
Ok @mateoholman

**Comment by ishankbahl** [comment]
@bvaughn Can I take snapshot testing?

**Comment by bvaughn** [comment]
Sure @ishankbahl

**Comment by ishankbahl** [comment]
@bvaughn What outline do you suggest for this guide?

**Comment by bvaughn** [comment]
No clue 😄 Haven't thought about it any. Just migrated this issue from the old GitHub repo.

**Comment by mateoholman** [comment]
Created a new PR for the "Publishing a Component" guide - #211.

**Comment by mateoholman** [comment]
I'll try to tackle the community section per the recommendations in  facebook/react#7117 next if no one is currently working on it?

**Comment by bvaughn** [comment]
It's yours, @mateoholman

**Comment by mateoholman** [comment]
After doing some prep/research, I see two approaches to the community section. Adding a new Resources section to the docs page similar to the [React Native Docs](https://facebook.github.io/react-native/docs/more-resources.html) or updating the [React Wiki](https://github.com/facebook/react/wiki). Both have been suggested in previous issues - any preference @bvaughn ?

**Comment by bvaughn** [comment]
👋  Hi @mateoholman. Sorry for the slow response time but I've been traveling.

I'm personally more a fan of the community page/section because it's more open for community contributions.

**Comment by mateoholman** [comment]
:wave: Hi @bvaughn . No worries, I had been scrubbing some of the outdated / broken links from the previous section. I should have a PR ready for review in a few days. :beers:

**Comment by mateoholman** [comment]
@bvaughn Created a PR for the updated community section  [PR #252 ](https://github.com/reactjs/reactjs.org/pull/252). Mind if I take the "Animation with React Motion" guide next?

**Comment by bvaughn** [comment]
Sure @mateoholman. It's yours.

**Comment by chiangs** [comment]
@bvaughn I'd like to take the Troubleshooting Guide:

Besides the following you listed (like wrong element casing, not loaded DOM container, missing React in scope) any other items you suggest being covered?

**Comment by bvaughn** [comment]
Okay @chiangs, please feel free to work on that! :)

**Comment by alDuncanson** [comment]
It looks like portals have been taken care of right?

https://reactjs.org/docs/portals.html

**Comment by spiritanand** [comment]
Is this issue open? Can you assign it to me if it is?

**Comment by rickhanlonii** [comment]
Closing as this issue was for the old docs and no one is tracking this now. Thanks all for your work!
